### PR TITLE
[DUOS-1763][risk=no] Refactor deprecated update method

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
Continued partial work towards completing https://broadworkbench.atlassian.net/browse/DUOS-1763

## Changes
* Minor name refactoring and removal of deprecated methods
* Non-trivial change to ElectionService to remove a call that updates DARs.
  * I've maintained the current functionality of email for disabled datasets
  * I removed the modifications to the DAR when that is the case, however. These should trigger a manual intervention.
* Fixes a bug introduced with https://github.com/DataBiosphere/consent/pull/1637 where draft dars with multiple datasets are created with only the first dataset in the list of dataset ids passed in from the UI.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/service/ElectionService.java [hunk 1]
```diff
@@ -419,51 +422,32 @@ private Election validateAndGetDULElection(String referenceId, ElectionType elec
...
-            DatasetDetailEntry entry = new DatasetDetailEntry();
-            entry.setDatasetId(dataSet.getDataSetId().toString());
-            entry.setName(dataSet.getName());
-            entry.setObjectId(dataSet.getObjectId());
-            activeDatasetDetailEntries.add(entry);
-        });
-        dar.setDatasetIds(activeDatasetIds);
-        dar.getData().setDatasetDetail(activeDatasetDetailEntries);
-        dataAccessRequestService.updateByReferenceId(referenceId, dar.getData());
     }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/ElectionService.java:451]: I'm removing this method as it is very problematic. This code path was hit when an election is created for DARs that referred to disabled datasets. This method modified the DAR directly outside of the normal DAR update process and is initiated by either an admin or chairperson. All DAR updates need to happen through a DAR update path, not an election creation path.

File: src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java [hunk 2]
```diff
@@ -247,8 +247,9 @@ public Response updatePartialDataAccessRequest(
       // it in dar data.
       data.setReferenceId(originalDar.getReferenceId());
       originalDar.setData(data);
+      originalDar.addDatasetIds(data.getDatasetIds());
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java:250]: Fixes a bug introduced in https://github.com/DataBiosphere/consent/pull/1637

File: src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java [hunk 3]
```diff
@@ -403,8 +404,7 @@ private DataAccessRequest populateDarFromJsonString(User user, String json) {
       data.setReferenceId(referenceId);
     }
     newDar.setData(data);
-    // TODO: Update the UI to pass in dataset ids on the DAR object itself.
-    newDar.setDatasetIds(data.getDatasetIds());
+    newDar.addDatasetIds(data.getDatasetIds());
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java:407]: Fixes a bug introduced in https://github.com/DataBiosphere/consent/pull/1637

File: src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java [hunk 4]
```diff
@@ -160,7 +145,7 @@ void updateDataByReferenceId(
   @RegisterArgumentFactory(JsonArgumentFactory.class)
   @SqlUpdate(
       "UPDATE data_access_request SET data = to_jsonb(:data), user_id = :userId, sort_date = :sortDate, submission_date = :submissionDate, update_date = :updateDate WHERE reference_id = :referenceId")
-  void updateDataByReferenceIdVersion2(
+  void updateDataByReferenceId(
```
Comment (lu-c) [inline src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java:148]: Why are we getting rid of the versioning?

**Review by lu-c — APPROVED** [review]
Looks good!

File: src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java (hunk 4 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java:148]: There were two versions of `update` previous to this PR. I was able to refactor away the deprecated version to use the newer version. This just removes the outdated `Version2` in the name.

File: src/test/java/org/broadinstitute/consent/http/db/DataAccessRequestDAOTest.java [hunk 5]
```diff
@@ -123,18 +123,6 @@ public void testFindByReferenceIds() {
...
-        dar.getData().setRus(rus);
-        dar.getData().setValidRestriction(false);
-        dataAccessRequestDAO.updateDataByReferenceId(dar.getReferenceId(), dar.getData());
-        DataAccessRequest updatedDar = dataAccessRequestDAO.findByReferenceId(dar.getReferenceId());
-        assertEquals(rus, updatedDar.getData().getRus());
-        assertFalse(updatedDar.getData().getValidRestriction());
-    }
-
     @Test
     public void testUpdateByReferenceIdVersion2() {
```
Comment (shaemarks) [inline src/test/java/org/broadinstitute/consent/http/db/DataAccessRequestDAOTest.java]:
```suggestion
    public void testUpdateByReferenceId() {
```

File: src/main/java/org/broadinstitute/consent/http/service/ElectionService.java [hunk 6]
```diff
@@ -419,51 +422,32 @@ private Election validateAndGetDULElection(String referenceId, ElectionType elec
...
+            }
         }
         return consentElection;
     }
 
-    private List<Dataset> verifyActiveDataSets(DataAccessRequest dar, String referenceId) throws Exception {
-        List<Integer> dataSets = Objects.nonNull(dar) && Objects.nonNull(dar.getData()) ? dar.getDatasetIds() : Collections.emptyList();
-        List<Dataset> dataSetList = dataSets.isEmpty() ? Collections.emptyList() : dataSetDAO.findDatasetsByIdList(dataSets);
-        List<String> disabledDataSets = dataSetList.stream()
+    private void verifyActiveDataSets(DataAccessRequest dataAccessRequest, List<Integer> datasetIds) throws Exception {
```
Comment (shaemarks) [inline src/main/java/org/broadinstitute/consent/http/service/ElectionService.java]:
```suggestion
    private void verifyActiveDatasets(DataAccessRequest dataAccessRequest, List<Integer> datasetIds) throws Exception {
```

File: src/main/java/org/broadinstitute/consent/http/service/ElectionService.java [hunk 7]
```diff
@@ -419,51 +422,32 @@ private Election validateAndGetDULElection(String referenceId, ElectionType elec
             if (Objects.isNull(dataAccessRequest) || Objects.isNull(dataAccessRequest.getData())) {
                 throw new NotFoundException();
             }
-            List<Dataset> dataSets = verifyActiveDataSets(dataAccessRequest, referenceId);
-            Consent consent = consentDAO.findConsentFromDatasetID(dataSets.get(0).getDataSetId());
-            consentElection = electionDAO.findLastElectionByReferenceIdAndStatus(consent.getConsentId(), ElectionStatus.CLOSED.getValue());
+            List<Integer> datasetIds = dataAccessRequestDAO.findDARDatasetRelations(referenceId);
+            if (!datasetIds.isEmpty()) {
+                verifyActiveDataSets(dataAccessRequest, datasetIds);
```
Comment (shaemarks) [inline src/main/java/org/broadinstitute/consent/http/service/ElectionService.java]:
```suggestion
                verifyActiveDatasets(dataAccessRequest, datasetIds);
```

**Review by shaemarks — APPROVED** [review]
I left a couple of naming suggestions, but looks good to me! Can confirm that the bug with not all datasets being saved on drafts has been resolved

**Pull request merged**